packages/atomic/src/utils/debounce-utils.tsx

```diff
@@ -0,0 +1,52 @@
+export interface DebouncedQueueOptions {
+  delay: number;
+}
+
+export interface DebouncedQueue {
+  enqueue(execute: () => void, uniqueId?: string): void;
+  clear(): void;
+  cancelActionIfQueued(id: string): void;
+}
+
+interface QueuedAction {
+  id?: string;
+  execute: () => void;
+}
+
+export function buildDebouncedQueue(
+  options: DebouncedQueueOptions
+): DebouncedQueue {
+  let actions: QueuedAction[] = [];
+  let intervalId: ReturnType<typeof setInterval> | null = null;
+
+  function dequeueAction() {
+    const action = actions.shift();
+    if (action) {
+      action.execute();
+    } else {
+      clearInterval(intervalId!);
+      intervalId = null;
```

Comment:
The interval is only cleared once it's triggered on an empty list. That's on purpose.